app/controllers/settings/general.js

```diff
@@ -134,6 +134,44 @@ export default Controller.extend({
             }
         },
 
+        toggleLeaveSettingsModal(transition) {
+            let leaveTransition = this.get('leaveSettingsTransition');
+
+            if (!transition && this.get('showLeaveSettingsModal')) {
+                this.set('showLeaveSettingsModal', false);
+                return;
+            }
+
+            if (!leaveTransition || transition.targetName === leaveTransition.targetName) {
+                this.set('leaveSettingsTransition', transition);
+
+                // if a save is running, wait for it to finish then transition
+                if (this.get('saveTasks.isRunning')) {
```

Comment:
This has been copied directly from the editor screen but won't work here because the `settings/general` controller does not have a `saveTasks` task group.

Reply:
@AileenCGN sorry, I wasn't very clear in my message yesterday 😞 There wasn't any need to remove this, preventing the transition until the save is complete is still useful so that errors can be displayed and there's no chance the task will be cancelled by the transition.

There are two approaches for a fix:

1. add the code back and change `get('saveTasks.isRunning'`) to `get('save.isRunning')`
2. add the code back and also add a ` saveTasks` task group that has the `save` task in it

Reply:
Gotcha! Was thinking about option 1, but then not sure... 😬